iroha_p2p/src/peer.rs

```diff
@@ -403,13 +405,115 @@ impl Garbage {
-            let _ = stream.read_exact(&mut buf).await?;
-        } else {
+        if size >= MAX_HANDSHAKE_LENGTH {
             return Err(Error::Handshake);
         }
+        // Reading garbage
+        let mut buf = vec![0_u8; size];
+        let _ = stream.read_exact(&mut buf).await?;
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::{fmt::Debug, time::Duration};
+
+    use iroha_actor::Addr;
+    use tokio::{
+        net::TcpListener,
+        sync::mpsc::{self, Receiver},
+        time,
+    };
+    use ursa::{encryption::symm::chacha20poly1305::ChaCha20Poly1305, kex::x25519::X25519Sha256};
+
+    use super::*;
+    use iroha_logger::config::LoggerConfiguration;
+
+    #[derive(iroha_actor::Message, Clone, Debug, Decode, Encode)]
+    struct TestMessage(String);
+
+    fn peer_id() -> PeerId {
+        let port = unique_port::get_unique_free_port().unwrap();
+        let address = format!("127.0.0.1:{}", port);
+        PeerId {
+            address,
+            public_key: None,
+        }
+    }
+
+    fn peer_to<T, K, E>(id: PeerId) -> (Peer<T, K, E>, Receiver<Received<T>>)
+        where
+            T: Encode + Decode + Send + Clone + 'static + Debug,
+            K: KeyExchangeScheme + Send + 'static,
+            E: Encryptor + Send + 'static,
+    {
+        let (sender, receiver) = mpsc::channel(100);
+        (Peer::new_to(id, sender.into()).unwrap(), receiver)
+    }
+
+    fn peer_from<T, K, E>(id: PeerId, stream: TcpStream) -> (Peer<T, K, E>, Receiver<Received<T>>)
+        where
+            T: Encode + Decode + Send + Clone + 'static + Debug,
+            K: KeyExchangeScheme + Send + 'static,
+            E: Encryptor + Send + 'static,
+    {
+        let (sender, receiver) = mpsc::channel(100);
+        (Peer::new_from(id, stream, sender.into()).unwrap(), receiver)
+    }
+
+    async fn peer_ping_pong<T, K, E>() -> (
+        (Addr<Peer<T, K, E>>, Receiver<Received<T>>),
+        (Addr<Peer<T, K, E>>, Receiver<Received<T>>),
+    )
+        where
+            T: Encode + Decode + Send + Clone + 'static + Debug,
+            K: KeyExchangeScheme + Send + 'static,
+            E: Encryptor + Send + 'static,
+    {
+        let id = peer_id();
+        let listener = TcpListener::bind(&id.address).await.unwrap();
+        let jh = tokio::spawn(async move {
+            let (stream, _) = listener.accept().await.unwrap();
+            stream
+        });
+
+        let (to, to_recv) = peer_to(id.clone());
+        let to = to.start().await;
+        tokio::time::sleep(Duration::from_millis(100)).await;
+
+        let (from, from_recv) = peer_from(id, jh.await.unwrap());
+        let from = from.start().await;
+        tokio::time::sleep(Duration::from_millis(100)).await;
+
+        ((to, to_recv), (from, from_recv))
+    }
+
+    #[tokio::test]
+    async fn simple_dimple() {
+
+        #[allow(unused)]
+        let telemetry = iroha_logger::init(LoggerConfiguration::default()).unwrap();
```

Comment:
```rust
drop(iroha_logger::init(LoggerConfiguration::default()));
```

Reply:
You can just drop it. Without using allow(unused)

Reply:
@eadventurous what are your thoughts on removing unused lint? it is kinda annoying and useless. We can make some types explicitly must to be used (like result), but for the rest is it worth it?

Reply:
I think it is worth it. It saved my life many times :)

I mean it's late, you are finalizing your draft and somewhere forget to complete a part of the code and here it is - always helps!

Reply:
I like that you need to explicitly drop or prefix with _. It adds clarity of the intent.

Reply:
Yeah, but it is quite hard to read though. And there are strange conventions, like:
```
let _ = get_not_copy(); // Will trigger clippy
drop(get_not_copy()); // Wont trigger clippy
```

Sometimes you want to just implicitly drop.

Reply:
Generally it is about good api design to my mind. You should get idea of that is returned and read it out of context, even if it is implicitly dropped.

Though `#[must_use]` can be explicitly added always

Reply:
We can drop this lint, because the compiler shows warnings on such things.

Reply:
Like we have our client api:
```rust
fn send_transaction_and_wait(&self, tx: Transaction) -> Result<Hash>;
```

Most of times you wont need a hash of transaction in this case. Like what can you do with it very usefull?

While using this method
```rust
fn send_transaction(&self, tx: Transaction) -> Result<Hash>;
```

You can use it somehow (like wait for finalization). But generally not sure if someone will need it still.

Reply:
@Revertron it shows just for types which are `#[must_use]`, like result and option.